### PR TITLE
Fixes sending topics when clicking anywhere in chat.

### PR DESCRIPTION
**Pull request by AnturK**
So yes, turns out this was an issue. It would bubble up to document body and try to use it as link.

### CODE REVIEW

**Comment by Mothblocks** [comment]
an unassuming pr that may or may not fix the reconnect bug :eyes: :eyes: :eyes:

**Comment by stylemistake** [comment]
Committed an alternative solution, moves target checking logic to one place.

File: tgui/packages/tgui/links.js [hunk 1]
```diff
@@ -13,17 +13,17 @@ export const captureExternalLinks = () => {
     /** @type {HTMLElement} */
     let target = e.target;
     // Recurse down the tree to find a valid link
-    while (target && target !== document.body) {
+    while (true) {
+      // Reached the end, bail.
+      if (!target || target === document.body) {
+        return;
+      }
```
Comment (coiax) [inline tgui/packages/tgui/links.js:20]: I don't quite understand why you can't move this if check into the conditional of the while loop

Comment (stylemistake) [reply tgui/packages/tgui/links.js:20]: It is necessary to bail based on this condition, not break the loop.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**